### ISSUE TITLE
Untracked todos

### ISSUE THREAD

**Issue by martin-lysk** [martin-lysk/collaborative-md#7]
> [!WARNING]  
> **42 TODO**'s without an issue reference found for this repository
>

`TODO`'s should **always** reference an issue in github to prevent them from getting lost.
Instead of writing a comment like `TODO fix this` that tent to get forgotten, reference an issue where the todo should be tackeled `TODO #42 fix this`.

<details><summary>Examples:</summary>

### The "getting to a happy path" TODO

You are Working on issue `#42` - on this awesome new feature. You are in flow staten and just want to see it working - yeah those exceptions are not helpful for now.

```javascript
try {
   // ... your crazy code bringing the new awesome feature of issue #42 to live
}  catch (e) {
   // TODO handle non happy path
}
```
 The `TODO` in code here is great to get it out of your head for **now** but it also tent to just end up forgotten.

Instead of loosing tack of the `TODO`'s just add your current issue number like:

```diff
try {
   // ... your crazy code bringing the new awesome feature of issue #42 to live
}  catch (e) {
-   // TODO handle non happy path
+   // TODO #42 handle non happy path
}
```

### The "ugly but we cant fix this for now - we should do this later" TODO

You are working on an issue that aims to make your app's UI/UX more inclusive. The `gender` property you need is sourced from a third party api and the servers response already contains it - but the library you are using to fetch the data didn't catch up on the types yet. Accessing `gender` is possible by casting the response to `any` and access the known property directly.

```typescript
...
// eslint-disable-next-line @typescript-eslint/no-explicit-any -- lib doesn't provide the gender property in its type yet 
const user = await thirdPartyLib.getUser() as any // TODO fix when library types are updated https://link.to.issue.in.third.party.lib
const name = user.name;
if (user.gender !== undefined) {
   ....
}
```
To pass the CI you deactivated the `no-explicit-any` lint rule for the this particular exception. Great you unblocked this issue from your dependency. You described the reason for the lint rule exception, the `TODO` you add describes what to do and even contains a link that keeps track if the problem can be fixed. Wow - this is a great - it contains all information needed - BUT again the chance is high that you will see this `TODO` only again when you had some hard time debugging a runtime error that was cause by an API change of the third party sdk, that replaced name by firstName and lastName property - and stays hidden by the `any` type.

How to deal with such a `TODO`? There is nothing you can do about this in your current issue `#42`. This time you want to track this TODO in a new issue. Just create an "remove any type when SDK XYZ provides gender property in the Type definition" Issue in github. And add the number of your newly created issue (lets say `#43`) to your TODO like:

```diff
...
// eslint-disable-next-line @typescript-eslint/no-explicit-any -- lib doesn't provide the gender property in its type yet 
-const user = await thirdPartyLib.getUser() as any // TODO fix when library types are updated 
+const user = await thirdPartyLib.getUser() as any // TODO #43 fix when library types are updated 
https://link.to.issue.in.third.party.lib
const name = user.name;
if (user.gender !== undefined) {
   ....
}
```

### The "premature optimization" TODO

You are still working on issue `#42` - you need to add a function that returns a user by its id. Users are stored in an array and finding it means iterating though the whole array. You have the feeling that this becomes slow in the future. You feel soo ashamed of not writing optimized code but something tells you - stop think about this - this is a premature optimization trap!

```typescript
const users: User[];
function getUserById(id: string) {
    // TODO O(n)... this iterates though all user we could consider to use a map here instead?
    return users.find(user => user.id === id)
}
```

The `TODO` you add gives you a better feeling because it expresses you are aware of the O(n) here and it can help you or your colleagues to tackle a possible performance issue. How to deal with such a `TODO`?
Think of it - might be you never have to change this - there might never be something `TODO` here - so just turn it into a `NOTE`:
```diff
const users: User[];
function getUserById(id: string) {
-   // TODO O(n)... this iterates though all user we could consider to use a map here instead?
+   // NOTE: O(n)... this iterates though all user we could consider to use a map here instead?
    return users.find(user => user.id === id)
}
```
</details>

### Todos without Issue Reference
</details>

<details open>
<summary><code>src/</code></summary>

<blockquote>

* [ ] [`runner.ts:2`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/runner.ts#L2): **Todo** } from './model/validation.js'
* [ ] [`runner.ts:120`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/runner.ts#L120): todo =&gt; todo.issueNumber === issueNr\)
<details open>
<summary><code>model/</code></summary>

<blockquote>

* [ ] [`model.repostate.ts:2`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.repostate.ts#L2): Todo } from './model.todo.js'
* [ ] [`model.todo.ts:1`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L1): Todo } from './validation.js'
* [ ] [`model.todo.ts:3`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L3): Todo {
* [ ] [`model.todo.ts:18`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L18): todo: TTodo\) {
* [ ] [`model.todo.ts:22`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L22): Todo: TTodo\) {
* [ ] [`model.todo.ts:29`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L29): Todo: TTodo\) {
* [ ] [`model.todostate.ts:1`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todostate.ts#L1): Todo } from './model.todo.js'
* [ ] [`validation.ts:5`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/validation.ts#L5): Todo = z.object\({
* [ ] [`validation.ts:15`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/validation.ts#L15): Todo = z.infer&lt;typeof zTodo&gt;
</blockquote>
</details>

<details open>
<summary><code>service/</code></summary>

<blockquote>

* [ ] [`comment.ts:1`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/comment.ts#L1): Todo } from '../model/validation.js'
* [ ] [`comment.ts:86`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/comment.ts#L86): todo of this.todos\) {
* [ ] [`datastore.ts:9`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L9): Todo } from 'model/model.todo.js'
* [ ] [`datastore.ts:97`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L97): todo of todos\) {
* [ ] [`datastore.ts:157`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L157): todo of subTree\) {
* [ ] [`datastore.ts:217`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L217): TODO fix this\\`, create an Issue for the problem on Github and reference the issue number: \\`TODO \#&#x2060;42 fix this\\`.
* [ ] [`github.ts:12`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/github.ts#L12): Todo } from '../model/model.todo.js'
</blockquote>
</details>

<details open>
<summary><code>util/</code></summary>

<blockquote>

* [ ] [`find-todo-stream.ts:2`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L2): Todo } from './todo-match.js'
* [ ] [`find-todo-stream.ts:3`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L3): Todo } from '../model/model.todo.js'
* [ ] [`find-todo-stream.ts:39`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L39): Todo
* [ ] [`find-todo-stream.ts:41`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L41): Todo = matchTodo\(line, this.issueNr?.toString\(\)\)
* [ ] [`todo-match.ts:27`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/todo-match.ts#L27): Todo = \(textLine: string, onlyIssueNumber?: string\): TodoRegexMatch | undefined =&gt; {
</blockquote>
</details>

</blockquote>
</details>

### Tracked Issues
* [Issue 57](57/#issuecomment-2063737916): *1* open TODOs
* [Issue 59](59/#issuecomment-2063738194): *1* open TODOs
* [Issue 62](62/#issuecomment-2063738035): *2* open TODOs
* [Issue 64](64/#issuecomment-2063738006): *1* open TODOs
* [Issue 65](65/#issuecomment-2063738060): *1* open TODOs
* [Issue 68](68/#issuecomment-2063737973): *2* open TODOs
* [Issue 70](70/#issuecomment-2063738085): *1* open TODOs
* [Issue 76](76/#issuecomment-2063738216): *1* open TODOs
* [Issue 77](77/#issuecomment-2063738119): *2* open TODOs
* [Issue 80](80/#issuecomment-2063738175): *3* open TODOs
* [Issue 93](93/#issuecomment-2063737949): *4* open TODOs
* [Issue 110](110/#issuecomment-2076126157): *2* open TODOs
### Todos without Issue Reference
</details>

<details open>
<summary><code>src/</code></summary>

<blockquote>

* [ ] [`runner.ts:2`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/runner.ts#L2): Todo } from './model/validation.js'
* [ ] [`runner.ts:120`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/runner.ts#L120): todo =&gt; todo.issueNumber === issueNr\)
<details open>
<summary><code>model/</code></summary>

<blockquote>

* [ ] [`model.repostate.ts:2`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.repostate.ts#L2): Todo } from './model.todo.js'
* [ ] [`model.todo.ts:1`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L1): Todo } from './validation.js'
* [ ] [`model.todo.ts:3`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L3): Todo {
* [ ] [`model.todo.ts:18`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L18): todo: TTodo\) {
* [ ] [`model.todo.ts:22`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L22): Todo: TTodo\) {
* [ ] [`model.todo.ts:29`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todo.ts#L29): Todo: TTodo\) {
* [ ] [`model.todostate.ts:1`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/model.todostate.ts#L1): Todo } from './model.todo.js'
* [ ] [`validation.ts:5`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/validation.ts#L5): Todo = z.object\({
* [ ] [`validation.ts:15`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/model/validation.ts#L15): Todo = z.infer&lt;typeof zTodo&gt;
</blockquote>
</details>

<details open>
<summary><code>service/</code></summary>

<blockquote>

* [ ] [`comment.ts:1`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/comment.ts#L1): Todo } from '../model/validation.js'
* [ ] [`comment.ts:86`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/comment.ts#L86): todo of this.todos\) {
* [ ] [`datastore.ts:9`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L9): Todo } from 'model/model.todo.js'
* [ ] [`datastore.ts:97`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L97): todo of todos\) {
* [ ] [`datastore.ts:157`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L157): todo of subTree\) {
* [ ] [`datastore.ts:217`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/datastore.ts#L217): TODO fix this\\`, create an Issue for the problem on Github and reference the issue number: \\`TODO \#&#x2060;42 fix this\\`.
* [ ] [`github.ts:12`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/service/github.ts#L12): Todo } from '../model/model.todo.js'
</blockquote>
</details>

<details open>
<summary><code>util/</code></summary>

<blockquote>

* [ ] [`find-todo-stream.ts:2`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L2): Todo } from './todo-match.js'
* [ ] [`find-todo-stream.ts:3`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L3): Todo } from '../model/model.todo.js'
* [ ] [`find-todo-stream.ts:39`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L39): Todo
* [ ] [`find-todo-stream.ts:41`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/find-todo-stream.ts#L41): Todo = matchTodo\(line, this.issueNr?.toString\(\)\)
* [ ] [`todo-match.ts:27`](https://github.com/todos-in/todohub/blob/d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf/src/util/todo-match.ts#L27): Todo = \(textLine: string, onlyIssueNumber?: string\): TodoRegexMatch | undefined =&gt; {
</blockquote>
</details>

</blockquote>
</details>

---
<sub>**Last updated:** d7fb7a0ce43fa92c5bd69562019519faf7dfc2bf</sub>
